portfolio/src/main/java/com/google/sps/servlets/ListComment.java

```diff
@@ -30,17 +30,19 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns comments. */
-@WebServlet("/addcomment")
-public class DataServlet extends HttpServlet {
+@WebServlet("/listcomment")
+public class ListComment extends HttpServlet {
 
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        int commentCount = getCommentCount(request);
         Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
         DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
         PreparedQuery results = datastore.prepare(query);
 
         List<String> comments = new ArrayList<>();
         for (Entity entity : results.asIterable()) {
+            if (comments.size() == commentCount) break;
```

Comment:
we can move this check a bit higher and return early so we don't have to even enter this for loop

Reply:
Discussed offline. I misread the purpose of this line.